### DISCUSSION TITLE
aero_opt.py: Cannot add leading edge constraints.  The dimensions of the  selected face are: (2, 2). For this case you must specify topID

### DISCUSSION THREAD

**Discussion by yangchas** [mdolab/MACH-Aero#156] [Q&A]
## Description
I am referring to the[ tutorial](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/opt_aero.html) and executing `python aero_opt.py`, but the following error occurred
`
raceback (most recent call last):
  File "aero_opt.py", line 133, in <module>
    DVCon.addLeTeConstraints(0, "iLow")
  File "/home/*/dafoam/packages/miniconda/lib/python3.8/site-packages/pygeo/constraints/DVCon.py", line 2001, in addLeTeConstraints
    raise Error(
baseclasses.utils.error.Error:
+------------------------------------------------------------------------------+
| Error: Cannot add leading edge constraints. One (and exactly one) of FFD     |
| block dimensions on the specified face must be 2. The dimensions of the      |
| selected face are: (2, 2). For this case you must specify topID              |
+------------------------------------------------------------------------------+
`
I am a beginner, could you please help me solve it！Thanks！

### Steps to reproduce issue

I am in directory` MACH-Aero-main/tutorial/opt/aero`, executing
`cp ../../airfoilopt/ffd/ffd.xyz .`
`cp ../../aero/analysis/wing_vol.cgns . `
`mpirun -np 1 python aero_opt.py`

### Code versions
adflow                 2.10.0             
asn1crypto             1.3.0              
certifi                2019.11.28         
cffi                   1.14.0             
cgnsutilities          2.6.0              
chardet                3.0.4              
conda                  4.8.2              
conda-package-handling 1.6.0              
cryptography           2.8                
Cython                 3.0a6              
dafoam                 3.1.1              
idna                   2.8                
idwarp                 2.6.0              
mdolab-baseclasses     1.6.1              
meson                  0.64.0             
mpi4py                 3.1.3              
multipoint             1.4.0              
numpy                  1.21.2             
petsc4py               3.14.1             
pip                    20.0.2             
pycosat                0.6.3              
pycparser              2.19               
pygeo                  1.12.2             
pyhyp                  2.5.0              
pyofm                  1.2.0              
pyOpenSSL              19.1.0             
pyoptsparse            2.10.1             
PySocks                1.7.1              
pyspline               1.5.2              
requests               2.22.0             
ruamel-yaml            0.15.87            
scipy                  1.7.1              
setuptools             45.2.0.post20200210
six                    1.14.0             
sqlitedict             2.1.0              
tqdm                   4.42.1             
urllib3                1.25.8             
wheel                  0.34.2

**Comment by hajdik** [comment]
If those are the exact commands you're running, you're probably copying the FFD from the airfoil tutorial. Check [here](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/machAeroTutorials/opt_ffd.html#files) for the correct file structure.
